### PR TITLE
AIP-84 Migrate test a connection to FastAPI API

### LINKED ISSUE #42594: AIP-84 Migrate the public endpoint Test Connection to FastAPI
### Description

Migrate the 'POST' test connection to Fast API.

### Use case/motivation

[AIP-84](https://cwiki.apache.org/confluence/display/AIRFLOW/AIP-84+UI+REST+API)

### Related issues

Relates to: https://github.com/apache/airflow/issues/42370

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by bugraoz93**
closes: #42594
It includes unifying parameter naming conventions for `test_connections.py`

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/api_fastapi/core_api/routes/public/connections.py [hunk 1]
```diff
@@ -180,3 +187,47 @@ async def patch_connection(
...
+        data = test_body.model_dump(by_alias=True)
+        data["conn_id"] = transient_conn_id
+        conn = Connection(**data)
+        os.environ[conn_env_var] = conn.get_uri()
+        test_status, test_message = conn.test_connection()
+        return ConnectionTestResponse.model_validate(
+            {"status": test_status, "message": test_message}, from_attributes=True
+        )
+    except ValidationError as err:
+        raise HTTPException(400, str(err.messages))
```
Comment (pierrejeambrun) [inline airflow/api_fastapi/core_api/routes/public/connections.py]: This is already handled by FastAPI server.

We can just keep the `finally` which is important.

(Natively ValidationError are 422, which is more accurate)

Comment (pierrejeambrun) [reply airflow/api_fastapi/core_api/routes/public/connections.py]: And you can remove the `400` from the doc.

**Review by pierrejeambrun — APPROVED** [review]
Nice.

One small change before we merge but looking good overall, thanks.

File: airflow/api_fastapi/core_api/routes/public/connections.py (hunk 1 above)
Comment (bugraoz93) [reply airflow/api_fastapi/core_api/routes/public/connections.py]: You are right. Missed that part. Removed them now.
Indeed, if I remove the `finally` that would mix everything up in the environment for the connections :)
Thanks for the quick review!

**Review by pierrejeambrun — DISMISSED** [review]

File: airflow/api_fastapi/core_api/routes/public/connections.py [hunk 2]
```diff
@@ -180,3 +186,44 @@ async def patch_connection(
...
+@connections_router.post(
+    "/test",
+    responses=create_openapi_http_exception_doc(
+        [
+            status.HTTP_401_UNAUTHORIZED,
+            status.HTTP_403_FORBIDDEN,
+        ]
+    ),
+)
+async def test_connection(
```
Comment (pierrejeambrun) [inline airflow/api_fastapi/core_api/routes/public/connections.py]: New convention, function must not be declared async.

More details here: 
https://github.com/apache/airflow/pull/43797

Can you remove it then we can merge ?

**Review by pierrejeambrun — COMMENTED** [review]
Small change

File: airflow/api_fastapi/core_api/routes/public/connections.py (hunk 2 above)
Comment (bugraoz93) [reply airflow/api_fastapi/core_api/routes/public/connections.py]: I’ll definitely make those changes. I came across this PR this morning and realized I missed it yesterday while pushing the documentation updates. Thanks for pointing it out!

Comment (bugraoz93) [reply airflow/api_fastapi/core_api/routes/public/connections.py]: Made the change.

**Review by amoghrajesh — APPROVED** [review]

**Review by pierrejeambrun — APPROVED** [review]
Good to merge, need rebasing to solve conflicts.

**Comment by bugraoz93** [comment]
Rebased and solved the conflicts.

**Pull request merged**